.github/workflows/pulsar-ci.yaml

```diff
@@ -0,0 +1,783 @@
+  ARTIFACT_RETENTION_DAYS: 3
+
+jobs:
+  changed_files_job:
+    name: 'Changed files check'
+    runs-on: ubuntu-20.04
+    outputs:
+      docs_only: ${{ steps.check_changes.outputs.docs_only }}
+      changed_tests: ${{ steps.changes.outputs.tests_files }}
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: Detect changed files
+        id:   changes
+        uses: apache/pulsar-test-infra/paths-filter@master
+        with:
+          filters: .github/changes-filter.yaml
+          list-files: csv
+
+      - name: Check changed files
+        id: check_changes
+        run: echo "::set-output name=docs_only::${{ fromJSON(steps.changes.outputs.all_count) == fromJSON(steps.changes.outputs.docs_count) && fromJSON(steps.changes.outputs.docs_count) > 0 }}"
+
+  build-and-license-check:
+    needs: changed_files_job
+    name: Build and License check
+    runs-on: ubuntu-20.04
+    timeout-minutes: 60
+
+    steps:
+      - name: checkout
+        if: ${{ needs.changed_files_job.outputs.docs_only != 'true' }}
+        uses: actions/checkout@v2
+
+      - name: Tune Runner VM
+        if: ${{ needs.changed_files_job.outputs.docs_only != 'true' }}
+        uses: ./.github/actions/tune-runner-vm
+
+      - name: Setup ssh access to build runner VM
+        # ssh access is enabled for builds in own forks
+        if: ${{ github.repository != 'apache/pulsar' && needs.changed_files_job.outputs.docs_only != 'true' }}
+        uses: ./.github/actions/ssh-access
+        with:
+          limit-access-to-actor: true
+
+      - name: Cache local Maven repository
+        if: ${{ needs.changed_files_job.outputs.docs_only != 'true' }}
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.m2/repository/*/*/*
+            !~/.m2/repository/org/apache/pulsar
+          key: ${{ runner.os }}-m2-dependencies-core-modules-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-m2-dependencies-core-modules-
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        if: ${{ needs.changed_files_job.outputs.docs_only != 'true' }}
+        with:
+          distribution: 'temurin'
+          java-version: 11
+
+      - name: Check source code license headers
+        if: ${{ needs.changed_files_job.outputs.docs_only != 'true' }}
+        run: mvn -B -T 8 -ntp initialize apache-rat:check license:check
+
+      - name: Build core-modules
+        if: ${{ needs.changed_files_job.outputs.docs_only != 'true' }}
+        run: |
+          mvn -B -T 1C -ntp -Pcore-modules,-main clean install -DskipTests -Dlicense.skip=true -Drat.skip=true
+
+      - name: Check binary licenses
+        if: ${{ needs.changed_files_job.outputs.docs_only != 'true' }}
+        run: src/check-binary-license.sh ./distribution/server/target/apache-pulsar-*-bin.tar.gz
```

Comment:
Hi @lhotari! Can we build full modules here? I notice an issue that `check-binary-license.sh` here doesn't check pulsar-presto-distribution's LICENSE because it's not in the binary with this workflow.

See also #16783. Since we don't upload any distribution binaries, I wonder why we build only core modules here.

Reply:
Covered by #15253